Tensile/KernelWriterAssembly.py

```diff
@@ -8663,6 +8922,24 @@ def computeStoreVgprs(self, kernel, divisor, tid0Scale, tid1Scale):
         kStr += inst("v_mul_lo_u32", vgpr(tmpV2), vgpr(tmpV1), sgpr(strideC1), "")
         kStr += inst("v_add_u32", vgpr(self.cinRowPtr), vgpr(tmpV2),vgpr(self.cinRowPtr),"rowStart VGPR")
         kStr += inst("v_add_u32", vgpr(tid1), vgpr(tmpV1),vgpr(tid1),"coord1 offset in MacroTile")
+        # subgroup[0] > 64 threads or MatrixInstB=1 
+        # requires to generate rowBlockId for each sIMD
+
+        tmpSgpr = self.getTmpSgpr(1)
+        # 1 row-block per Instruction
+        if  kernel["MatrixInstB"] == 1:
+          kStr += inst("v_lshrrev_b32", vgpr(tmpV4), log2(globalParameters["WavefrontWidth"]),vgpr(tid0),"Row-BlockId in macroTile0")
+          #kStr += inst("v_lshlrev_b32", vgpr(tmpV4), log2(kernel["MatrixInstM"]*kernel["ThreadTile0"]),vgpr(tmpV4),"Row-BlockId in macroTile0")
+          kStr += inst("s_mov_b32", sgpr(tmpSgpr), hex(kernel["MatrixInstM"]*kernel["ThreadTile0"]), "")
+          kStr += inst("v_mul_lo_u32", vgpr(tmpV4), sgpr(tmpSgpr),vgpr(tmpV4),"Row-BlockId in macroTile0")
+          kStr += inst("v_and_b32", vgpr(tid0), hex(globalParameters["WavefrontWidth"]-1), vgpr(tid0), "WaveFrontLaneId")
+       
+        # 2 row-blocks per Instruction
+        if kernel["MIWG0"] > globalParameters["WavefrontWidth"] and kernel["MatrixInstB"] != 1:
```

Comment:
elif could make this clearer to read

Reply:
changed to elif

Reply:
latest code brought back both version . there is subtle change in 2-row blocks per instruction